### ISSUE TITLE
Fix S3900: Constructor initializer false negative

### ISSUE THREAD

**Issue by valhristov** [SonarSource/sonar-csharp#1534]
### Description

The rule incorrectly suggests to refactor constructor by adding argument checks for arguments that are dereferenced in the this/base initializer. We should consider changing the rule message to suggest using defensive pattern in such case.

### Repro steps

```
public Program(int i) { }
public Program(string s) : this(s.Length) { }
//                              ^^^^^^^^ this is not raised, but should be
```

### Expected behavior

An issue is raised but not suggesting to add argument check, because it's impossible.

**Comment by pavel-mikula-sonarsource** [comment]
This requires to runt the SE for `base(...)` somehow. Is there even CFG available for it?

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1996](https://sonarsource.atlassian.net/browse/NET-1996)

[NET-1996]: https://sonarsource.atlassian.net/browse/NET-1996?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
